### PR TITLE
bip-0340: clarify that lift_x fails with out-of-range inputs

### PR DESCRIPTION
**Pull request by jonasnick**
Without this commit, it's not defined what happens if x is not in range 0..p-1.
However, lift_x may easily be called with out of range values. The reference
implementation of lift_x correctly returns failure in such cases.

CC: @real-or-random @sipa

### CODE REVIEW

File: bip-0340.mediawiki [hunk 1]
```diff
@@ -109,8 +109,9 @@ The following conventions are used, with constants as defined for [https://www.s
...
 ** The function ''has_even_y(P)'', where ''P'' is a point for which ''not is_infinite(P)'', returns ''y(P) mod 2 = 0''.
-** The function ''lift_x(x)'', where ''x'' is an integer in range ''0..p-1'', returns the point ''P'' for which ''x(P) = x''<ref>
-    Given a candidate X coordinate ''x'' in the range ''0..p-1'', there exist either exactly two or exactly zero valid Y coordinates. If no valid Y coordinate exists, then ''x'' is not a valid X coordinate either, i.e., no point ''P'' exists for which ''x(P) = x''. The valid Y coordinates for a given candidate ''x'' are the square roots of ''c = x<sup>3</sup> + 7 mod p'' and they can be computed as ''y = &plusmn;c<sup>(p+1)/4</sup> mod p'' (see [https://en.wikipedia.org/wiki/Quadratic_residue#Prime_or_prime_power_modulus Quadratic residue]) if they exist, which can be checked by squaring and comparing with ''c''.</ref> and ''has_even_y(P)'', or fails if no such point exists. The function ''lift_x(x)'' is equivalent to the following pseudocode:
+** The function ''lift_x(x)'', where ''x'' is an integer in range ''0..2<sup>256</sup>-1'', returns the point ''P'' for which ''x(P) = x''<ref>
```
Comment (real-or-random) [inline bip-0340.mediawiki:112]: nit:
```suggestion
** The function ''lift_x(x)'', where ''x'' is a 256-bit unsigned integer, returns the point ''P'' for which ''x(P) = x''<ref>
```

Simpler and consistent with two lines above. Or just "unsigned integer" would also be fine, that's even simpler.

Comment (jonasnick) [reply bip-0340.mediawiki]: The size of the input is a helpful hint for implementors. Therefore went with your original suggestion.

**Review by real-or-random — APPROVED** [review]
ACK 0144413e91bf5d71b54126783acaf339422abe7e

**Comment by sipa** [comment]
ACK 0144413e91bf5d71b54126783acaf339422abe7e

**Pull request merged**